### PR TITLE
db-model: Added connecting to a database

### PR DESCRIPTION
**Pull request by wojtekbu**
* added model calsses
* added insert service

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,6 +28,17 @@
 			<artifactId>spring-boot-starter</artifactId>
 		</dependency>
 
+        <dependency>
```
Comment (pgesek) [inline pom.xml]: Bad identation

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,6 @@
+spring.jpa.hibernate.ddl-auto=update
+spring.datasource.url=jdbc:mysql://localhost:3306/wojtech?useSSL=false
```
Comment (pgesek) [inline src/main/resources/application.properties]: change the db

File: src/main/java/org/openmrs/isanteplus/performancedata/model/connection/Inserter.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    private final MySqlConnector connector;
+
+    public Inserter(GeneratorOptions options) {
+        this.connector = new MySqlConnector(options.getDbLogin(), options.getDbPassword(),
+                options.getDbServer(), options.getDbPort());
+    }
+
+    public void insertEntities(List<AbstractEntity> entities) throws SQLException {
+            Connection connection = connector.getConnection();
```
Comment (pgesek) [inline src/main/java/org/openmrs/isanteplus/performancedata/model/connection/Inserter.java]: ideally close should be done in a finally block, or try with resource should be sued

File: src/main/java/org/openmrs/isanteplus/performancedata/model/Encounter.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        preparedSql = "insert into encounter"
+                + " (encounter_id, encounter_type, parent_id, location_id, form_id, " +
+                "encounter_datetime, creator, date_created, voided, voided_by, date_voided, " +
+                "void_reason, changed_by, date_changed, visit_id, uuid)"
+                + " values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)";
+    }
+
+    @Override
+    public PreparedStatement fillPreparedStatement(PreparedStatement ps) throws SQLException {
+        ps.setString(1, Long.toString(encounterType));
```
Comment (pgesek) [inline src/main/java/org/openmrs/isanteplus/performancedata/model/Encounter.java]: This is pretty ugly, can't we just use Spring JDBC for this? 

https://www.mkyong.com/spring/spring-named-parameters-examples-in-simplejdbctemplate/ - Example 2

With a BeanPropertySqlParameterSource you can drop the whole prepare method and replace the question marks by property names like :name, :encounterType, etc.

File: src/main/java/org/openmrs/isanteplus/performancedata/model/connection/Inserter.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+            if (CollectionUtils.isEmpty(entities)) {
```
Comment (areklalo) [inline src/main/java/org/openmrs/isanteplus/performancedata/model/connection/Inserter.java]: I think you should use here negation.

File: pom.xml [hunk 6]
```diff
@@ -1,48 +1,72 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (wojtekbu) [inline pom.xml]: I have rewritten this file due to indentation issues.

**Pull request merged**